--- mysql-connector-c-6.0.2/vio/viosocket.c	2009-08-07 05:31:12.000000000 -0700
+++ mysql-connector-c-6.0.2-ribs/vio/viosocket.c	2012-06-15 18:48:04.272931403 -0700
@@ -21,6 +21,7 @@
 */
 
 #include "vio_priv.h"
+#include "ribs2.h"
 
 int vio_errno(Vio *vio __attribute__((unused)))
 {
@@ -41,7 +42,7 @@
   r = recv(vio->sd, buf, size,0);
 #else
   errno=0;					/* For linux */
-  r = read(vio->sd, buf, size);
+  r = ribs_read(vio->sd, buf, size);
 #endif /* __WIN__ */
 #ifndef DBUG_OFF
   if (r == (size_t) -1)
@@ -108,7 +109,7 @@
 #ifdef __WIN__
   r = send(vio->sd, buf, size,0);
 #else
-  r = write(vio->sd, buf, size);
+  r = ribs_write(vio->sd, buf, size);
 #endif /* __WIN__ */
 #ifndef DBUG_OFF
   if (r == (size_t) -1)
--- mysql-connector-c-6.0.2/libmysql/client.c	2009-08-07 05:31:12.000000000 -0700
+++ mysql-connector-c-6.0.2-ribs/libmysql/client.c	2012-06-15 18:48:26.644649439 -0700
@@ -37,6 +37,7 @@
 #include <my_global.h>
 
 #include "mysql.h"
+#include "ribs2.h"
 
 #ifndef __WIN__
 #include <netdb.h>
@@ -157,8 +158,8 @@
     exactly like the normal connect() call does.
   */
 
-  if (timeout == 0)
-    return connect(fd, (struct sockaddr*) name, namelen);
+//  if (timeout == 0)
+    return ribs_connect(fd, (struct sockaddr*) name, namelen, timeout);
 
   flags = fcntl(fd, F_GETFL, 0);	  /* Set socket to not block */
 #ifdef O_NONBLOCK
--- mysql-connector-c-6.0.2/include/ribs2.h	1969-12-31 16:00:00.000000000 -0800
+++ mysql-connector-c-6.0.2-ribs/include/ribs2.h	2012-06-15 18:42:46.216929429 -0700
@@ -0,0 +1,4 @@
+       int ribs_connect(int sockfd, const struct sockaddr *addr, socklen_t addrlen, int timeout);
+       ssize_t ribs_read(int fd, void *buf, size_t count);
+       ssize_t ribs_write(int fd, const void *buf, size_t count);
+
--- mysql-connector-c-6.0.2/./tests/CMakeLists.txt	2009-08-07 05:31:12.000000000 -0700
+++ mysql-connector-c-6.0.2-ribs/./tests/CMakeLists.txt	2012-06-15 19:15:24.048316011 -0700
@@ -17,5 +17,5 @@
 
 INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/include)
 
-ADD_EXECUTABLE(mysql_client_test mysql_client_test.c ../mysys/my_memmem.c)
-TARGET_LINK_LIBRARIES(mysql_client_test mysqlclient)
+# ADD_EXECUTABLE(mysql_client_test mysql_client_test.c ../mysys/my_memmem.c)
+#TARGET_LINK_LIBRARIES(mysql_client_test mysqlclient)
--- mysql-connector-c-6.0.2/./unittest/libmysql/CMakeLists.txt	2009-08-07 05:31:12.000000000 -0700
+++ mysql-connector-c-6.0.2-ribs/./unittest/libmysql/CMakeLists.txt	2012-06-15 19:16:50.675226832 -0700
@@ -24,9 +24,9 @@
 SET(API_TESTS "basic-t" "fetch" "charset" "logs" "errors" "cursor" "view" "ps" "ps_bugs" 
               "sp" "result" "connection" "misc")
 
-FOREACH(API_TEST ${API_TESTS})
-  ADD_EXECUTABLE(${API_TEST} ${API_TEST}.c)
-  TARGET_LINK_LIBRARIES(${API_TEST} mytap mysqlclient)
-  ADD_TEST(${API_TEST} ${EXECUTABLE_OUTPUT_PATH}/${API_TEST})
-  SET_TESTS_PROPERTIES(${API_TEST} PROPERTIES TIMEOUT 120)
-ENDFOREACH(API_TEST)
+#FOREACH(API_TEST ${API_TESTS})
+#  ADD_EXECUTABLE(${API_TEST} ${API_TEST}.c)
+#  TARGET_LINK_LIBRARIES(${API_TEST} mytap mysqlclient)
+#  ADD_TEST(${API_TEST} ${EXECUTABLE_OUTPUT_PATH}/${API_TEST})
+#  SET_TESTS_PROPERTIES(${API_TEST} PROPERTIES TIMEOUT 120)
+#ENDFOREACH(API_TEST)
--- mysql-connector-c-6.0.2/./unittest/examples/CMakeLists.txt	2009-08-07 05:31:12.000000000 -0700
+++ mysql-connector-c-6.0.2-ribs/./unittest/examples/CMakeLists.txt	2012-06-15 19:16:16.947650970 -0700
@@ -19,20 +19,20 @@
                     ${CMAKE_SOURCE_DIR}/regex
                     ${CMAKE_SOURCE_DIR}/extra/yassl/include
                     ${CMAKE_SOURCE_DIR}/unittest/mytap)
-ADD_EXECUTABLE(simple-t simple-t.c)
-TARGET_LINK_LIBRARIES(simple-t mytap)
+#ADD_EXECUTABLE(simple-t simple-t.c)
+#TARGET_LINK_LIBRARIES(simple-t mytap)
 
-ADD_EXECUTABLE(skip-t skip-t.c)
-TARGET_LINK_LIBRARIES(skip-t mytap)
+#ADD_EXECUTABLE(skip-t skip-t.c)
+#TARGET_LINK_LIBRARIES(skip-t mytap)
 
-ADD_EXECUTABLE(todo-t todo-t.c)
-TARGET_LINK_LIBRARIES(todo-t mytap)
+#ADD_EXECUTABLE(todo-t todo-t.c)
+#TARGET_LINK_LIBRARIES(todo-t mytap)
 
-ADD_EXECUTABLE(skip_all-t skip_all-t.c)
-TARGET_LINK_LIBRARIES(skip_all-t mytap)
+#ADD_EXECUTABLE(skip_all-t skip_all-t.c)
+#TARGET_LINK_LIBRARIES(skip_all-t mytap)
 
-ADD_EXECUTABLE(no_plan-t no_plan-t.c)
-TARGET_LINK_LIBRARIES(no_plan-t mytap)
+#ADD_EXECUTABLE(no_plan-t no_plan-t.c)
+#TARGET_LINK_LIBRARIES(no_plan-t mytap)
 
-ADD_EXECUTABLE(core-t core-t.c)
-TARGET_LINK_LIBRARIES(core-t mytap)
+#ADD_EXECUTABLE(core-t core-t.c)
+#TARGET_LINK_LIBRARIES(core-t mytap)
--- mysql-connector-c-6.0.2/./mysys/CMakeLists.txt	2009-08-07 05:31:12.000000000 -0700
+++ mysql-connector-c-6.0.2-ribs/./mysys/CMakeLists.txt	2012-06-15 18:47:41.885210728 -0700
@@ -42,7 +42,8 @@
 
   INCLUDE_DIRECTORIES(${ZLIB_INCLUDE_DIR}
                       ${CMAKE_SOURCE_DIR}/include
-                      ${CMAKE_SOURCE_DIR}/mysys)
+                      ${CMAKE_SOURCE_DIR}/mysys
+                      ${CMAKE_SOURCE_DIR}/extlib/ribs2)
 
   ADD_LIBRARY(mysys ${MYSYS_SOURCES})
   TARGET_LINK_LIBRARIES(mysys ${CMAKE_THREAD_LIBS_INIT})
